### PR TITLE
update(UI): Refactor storage space UI

### LINKED ISSUE #780: component(files): Implement file storage amount
**OS:**
-All

**Task:**
-In the top of files it displays amount of storage used in Files. Right now it is not implemented so it says 0mb/10mb. We want to display how much the user has used

**Screenshot:**

image

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Add max and current size on Storage
image

### 2. Do different arch to handle with use_state and use_ref variable, test purpose 
image

- 

### Which issue(s) this PR fixes 🔨

- Resolve #780

### CODE REVIEW

File: ui/src/layouts/storage/functions.rs
Comment (dariusc93) [inline ui/src/layouts/storage/functions.rs]: I cant recall but i believe this import is only compatible with unix-like systems. Could you confirm this works on windows too?

Comment (lgmarchi) [reply ui/src/layouts/storage/functions.rs]: Hum... I think you are right, I didn't test on Windows.

Comment (dariusc93) [reply ui/src/layouts/storage/functions.rs]: Might want to look into `sysinfo` crate to see if that can add any help with this, which should work across platforms

Comment (lgmarchi) [reply ui/src/layouts/storage/functions.rs]: sysinfo may work, but I decided to not use, because it had some unsafe code, but I will take a look again

**Comment by phillsatellite** [comment]
I also tried compiling on Windows and didnt have any luck I tried doing cargo clean before so as well 

Screenshot 2023-05-09 at 12 25 14 PM

**Comment by lgmarchi** [comment]
Draft as now, waiting new update from Warp.

File: common/src/warp_runner/manager/commands/constellation_commands.rs
Comment (dariusc93) [inline common/src/warp_runner/manager/commands/constellation_commands.rs]: No need for mutable borrow if the calls are immutable
```suggestion
fn get_storage_size(warp_storage: &warp_storage) -> Result<(usize, usize), Error> {
```

**Comment by dariusc93** [comment]
~~Note: This PR is blocked until #808 is merged~~

**Comment by lgmarchi** [comment]
> Note: This PR is blocked until #808 is merged

PR Merged!

**Review by dariusc93 — APPROVED** [review]

File: common/locales/en-US/main.ftl
Comment (stavares843) [inline common/locales/en-US/main.ftl]:
```suggestion
    .storage-current-size = Current Space:
```

**Review by stavares843 — COMMENTED** [review]
we have 1gb space, when we upload a file of 2gb there's an error in terminal but no visible error to the user


```
2023-05-12 17:43:42 | ERROR | Error when upload file: InvalidLength { context: "DJI Osmo Pocket downloadable 4K video sample test footage file - 2.MP4", current: 2168979744, minimum: None, maximum: Some(1073741824) }
``` 

https://github.com/Satellite-im/Uplink/assets/29093946/f381d72c-2be0-419d-9df3-bd125df57489





macOS, m1

**Comment by stavares843** [comment]
if you need big files for testing purposes
https://www.mediafire.com/folder/9971h5184vpju/Osmo_Pocket_test_files

**Review by stavares843 — COMMENTED** [review]
also there was this glitch when changing tabs maybe we could fetch the storage without re-render the labels


https://github.com/Satellite-im/Uplink/assets/29093946/dd5072ea-7cfe-4823-ba93-11254b3189da

**Comment by sdwoodbury** [comment]
@stavares843  given that this is just a refactor and Lucas is out of office at the moment, it is probably is best to make tickets for outstanding issues and approve this PR.

**Pull request merged**
